### PR TITLE
Adding market data

### PR DESCRIPTION
**Pull request by supersecretcatia**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -58,6 +48,15 @@
             <artifactId>springfox-swagger-ui</artifactId>
             <version>${springfox-swagger.version}</version>
         </dependency>
+
+        <!-- Normal Distribution -->
+        <dependency>
+            <groupId>org.apache.commons</groupId>
+            <artifactId>commons-math3</artifactId>
+            <version>3.6.1</version>
```
Comment (supersecretliam) [inline pom.xml]: Maybe add this as a property

File: pom.xml [hunk 2]
```diff
@@ -72,6 +71,15 @@
             <type>pom</type>
             <scope>import</scope>
         </dependency>
+        <dependency>
```
Comment (supersecretliam) [inline pom.xml:75]: why are these being added again? We already have them in the pom

File: src/main/java/com/iggroup/universityworkshopmw/domain/model/Market.java [hunk 3]
```diff
@@ -0,0 +1,16 @@
+package com.iggroup.universityworkshopmw.domain.model;
+
+import lombok.Builder;
+import lombok.Getter;
+import lombok.ToString;
+
+@ToString
```
Comment (supersecretliam) [inline src/main/java/com/iggroup/universityworkshopmw/domain/model/Market.java]: Maybe we could add @EqualsAndHashCode on the domain objects too (as it's normally included with Data)

File: src/main/java/com/iggroup/universityworkshopmw/domain/services/PriceGeneratorService.java [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+   public PriceGeneratorService(MarketDataService marketDataService) {
+      this.marketDataService = marketDataService;
+   }
+
+   @Scheduled(fixedRate = 1000)
+   private void updateMarketPrices() {
+      List<Map.Entry<String, Market>> updateSubList = marketDataService.getShuffledMapSubset();
+      List<String> marketIdsUpdated = new ArrayList<>();
+
+      for (Map.Entry<String, Market> entry : updateSubList) {
```
Comment (supersecretliam) [inline src/main/java/com/iggroup/universityworkshopmw/domain/services/PriceGeneratorService.java]: Maybe we could use a stream here, just to be consistent

**Review by supersecretliam — COMMENTED** [review]
Looks really good. Nice work!

File: pom.xml (hunk 2 above)
Comment (supersecretcatia) [reply pom.xml:75]: I moved them to the bottom to organise the pom. Git just shows it as being removed near the top and added here :)

Comment (supersecretliam) [reply pom.xml:75]: My bad!

**Review by supersecretliam — APPROVED** [review]
Nice work

**Pull request merged**